src/controllers/projects.js

```
@@ -1,177 +1,97 @@
+    return res.status(500).json({
+      error: true,
     });
   }
-});
+};
 
-router.post('/', (req, res) => {
-  let errors = '';
-  const newProject = req.body;
-  const checkData = (data, err) => {
-    if (!newProject[data]) {
-      errors += err;
+const getProjectById = async (req, res) => {
+  const isValidObjectId = (id) => {
+    if (ObjectId.isValid(id)) {
+      if ((String)(new ObjectId(id)) === id) {
+        return true;
+      }
     }
+    return false;
   };
-  newProject.id = (projects.length + 1).toString();
-  projects.push(newProject);
-  checkData('projectName', 'Project needs a name. ');
-  checkData('description', 'Project needs a description. ');
-  checkData('startDate', 'Project needs a start date. ');
-  newProject.status = newProject.endDate ? newProject.status : 'active';
-  checkData('status', 'Project needs a status. ');
-  if (errors === '') {
-    fs.writeFile('./src/data/projects.json', JSON.stringify(projects, null, 2), (err) => {
-      if (err) {
-        res.status(400).json({
-          success: false,
-        });
-        return;
-      }
-      res.status(201).json({
-        success: true,
-        msg: 'Project created successfully',
-        data: newProject,
+  try {
+    const { id } = req.params;
+    if (!isValidObjectId(req.params.id)) {
+      return res.status(400).json({
+        message: `Invalid id: ${req.params.id}`,
+        data: undefined,
+        error: true,
+      });
+    }
+    const project = await Projects.findById(id);
+    if (!project) {
+      return res.status(404).json({
+        msg: `Couldnt find project with id ${id}`,
+        data: undefined,
+        error: true,
       });
+    }
+    return res.status(200).json({
+      msg: 'Project found succesfully',
+      data: project,
+      error: false,
     });
-  } else {
-    res.status(400).json({
-      success: false,
-      msg: errors,
+  } catch (error) {
+    return res.status(400).json({
+      msg: 'An error ocurred',
```

msg should be message and we should return the error in the message. Like this message: error